lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -454,6 +526,60 @@ downcastTxOut (TxOutInRecentEra addr val Alonzo.NoDatum Nothing)
 downcastTxOut (TxOutInRecentEra addr val (Alonzo.DatumHash dh) Nothing)
     = Right $ Alonzo.TxOut addr val (SJust dh)
 
+--
+-- MinimumUTxO
+--
+
+-- | Compute the minimum ada quantity required for a given 'TxOut'.
+--
+-- Unlike @Ledger.evaluateMinLovelaceOutput@, this function may return an
+-- overestimation for the sake of satisfying the property:
+--
+-- @
+--     forall out.
+--     let
+--         c = computeMinimumCoinForUTxO out
+--     in
+--         forall c' >= c.
+--         not $ isBelowMinimumCoinForTxOut modifyTxOutCoin (const c') out
+-- @
+--
+-- This makes it easy for callers to create outputs with near-minimum ada
+-- quantities regardless of the fact that modifying the ada 'Coin' value may
+-- itself change the size and min-ada requirement.
+computeMinimumCoinForTxOut
+    :: forall era. RecentEra era
+    -- FIXME [ADP-2353] Replace 'RecentEra' with 'IsRecentEra'
+    -> Core.PParams (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+    -> Coin
+computeMinimumCoinForTxOut era pp out = withCLIConstraint era $
+    evaluateMinLovelaceOutput pp (withMaxLengthSerializedCoin out)
+  where
+    withMaxLengthSerializedCoin
+        :: TxOut (ShelleyLedgerEra era)
+        -> TxOut (ShelleyLedgerEra era)
+    withMaxLengthSerializedCoin =
+        withStandardCryptoConstraint era $
+            modifyTxOutCoin era (const $ toLedger txOutMaxCoin)
+
+isBelowMinimumCoinForTxOut
+    :: forall era. RecentEra era
+    -- FIXME [ADP-2353] Replace 'RecentEra' with 'IsRecentEra'
+    -> Core.PParams (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+    -> Bool
+isBelowMinimumCoinForTxOut era pp out = withCLIConstraint era $
+    let
+        requiredMin = evaluateMinLovelaceOutput pp out
+        actualCoin = getCoin era out
+    in
+        actualCoin < requiredMin
+  where
```

Comment:
Hmm. This is just my opinion (so take it with a pinch of salt), but I still reckon using `where` would be clearer here.

In the suggestion below, the most important part of the function definition, namely the test `actualCoin < requiredMin`, is right at the top: (If the reader wishes to delve into more details, they can carry on reading.)

```suggestion
      actualCoin < requiredMin
  where
      requiredMin = evaluateMinLovelaceOutput pp out
      actualCoin = getCoin era out
```

Reply:
> the most important part of the function definition

🤔 Fair enough...